pubnub-kotlin/pubnub-kotlin-api/src/iosMain/kotlin/com/pubnub/api/PubNubImpl.kt

```diff
@@ -92,33 +103,59 @@ class PubNubImpl(override val configuration: PNConfiguration) : PubNub {
         replicate: Boolean,
         ttl: Int?
     ): Publish {
-        TODO("Not yet implemented")
-    }
-
+        return PublishImpl(
+            pubnub = pubNubObjC,
+            channel = channel,
+            message = message,
+            meta = meta,
+            shouldStore = shouldStore,
+            ttl = ttl
+        )
+    }
+
+    @OptIn(ExperimentalForeignApi::class)
+    // TODO: usePost parameter is not present in Swift SDK
     override fun fire(channel: String, message: Any, meta: Any?, usePost: Boolean, ttl: Int?): Publish {
-        TODO("Not yet implemented")
+        return PublishImpl(
+            pubnub = pubNubObjC,
+            channel = channel,
+            message = message,
+            meta = meta,
+            shouldStore = false,
+            ttl = 0
+        )
     }
 
+    @OptIn(ExperimentalForeignApi::class)
     override fun signal(channel: String, message: Any): Signal {
-        TODO("Not yet implemented")
+        return SignalImpl(pubnub = pubNubObjC, channel = channel, message = message)
     }
 
+    @OptIn(ExperimentalForeignApi::class)
     override fun getSubscribedChannels(): List<String> {
-        TODO("Not yet implemented")
+        return pubNubObjC.subscribedChannels() as List<String>
```

Comment:
also did you happen to check if this cast succeeds? (it probably does, but just wanted to know if you ran this code)

Reply:
FYI, I checked this casting and it works as expected